main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -614,12 +615,19 @@ object BuildServerProtocol {
     val thisProjectRef = Keys.thisProjectRef.value
     val thisConfig = Keys.configuration.value
     val scalaJars = Keys.scalaInstance.value.allJars.map(_.toURI.toString)
+    val jvmBuildTarget = javaHome.value match {
+      // TODO: populate Java version for forked java
+      case Some(forkedJava) => JvmBuildTarget(Some(forkedJava.toURI), None)
+      case None =>
+        JvmBuildTarget(Paths.get(sys.props("java.home")).toUri, sys.props("java.class.version"))
```

Comment:
Could it be that `java.home` or `java.class.version` are not defined?

Reply:
Thanks! Just refactored the logic to make it more robust